application/lib/tutorial/two/module/adjust_slider.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/semantics.dart';
+import 'dart:io' show Platform;
+
+class AdjustSliderState extends State<AdjustSlider> {
+  @override
+  double _currentValue = 0;
+  int stage = 1;
+  String displayText = "";
+
+  Widget build(BuildContext context) {
+    if (stage == 1 && _currentValue == 100) {
+      // Finished first part
+      stage = 2;
+    }
+    if (stage == 2 && _currentValue == 50) {
+      // Finished second part
+      stage = 3;
+    }
+    if (stage == 3 && _currentValue == 100) {
+      // Finished third part
+      stage = 4;
+    }
+    displayText = instructions(stage);
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text('Adjust slider page'),
+        ),
+        body: Center(
+            child: Column(children: [
+          Text(displayText),
+          Text(_currentValue.toString(), style: TextStyle(fontSize: 30)),
```

Comment:
```suggestion
          Text(_currentValue.toStringAsFixed(0), style: TextStyle(fontSize: 30)),
```

This prevents the floating point error that causes 30 to be displayed with many decimal places.